lib/auth/auth_with_roles.go

```diff
@@ -2946,15 +2989,19 @@ func (a *ServerWithRoles) DeleteNetworkRestrictions(ctx context.Context) error {
 	return a.authServer.DeleteNetworkRestrictions(ctx)
 }
 
-// TODO(awly): decouple auth.ClientI from auth.ServerWithRoles, they exist on
-// opposite sides of the connection.
+// GetMFADevices returns a list of MFA devices.
+func (a *ServerWithRoles) GetMFADevices(ctx context.Context, req *proto.GetMFADevicesRequest) (*proto.GetMFADevicesResponse, error) {
+	// Logged in users can view their own devices.
+	if req.GetTokenID() == "" {
+		req.Request = &proto.GetMFADevicesRequest_AuthenticatedUser{AuthenticatedUser: a.context.User.GetName()}
```

Comment:
Let's take the `AuthenticatedUser` out of the public API and pass it downwards in the stack through other means (using the context, for example).